stdlib/REPL/src/TerminalMenus/Pager.jl

```diff
@@ -0,0 +1,45 @@
+mutable struct Pager{C} <: _ConfiguredMenu{C}
+    lines::Array{String,1}
+    pagesize::Int
+    pageoffset::Int
+    selected::Nothing
+    config::C
+end
+
+function Pager(text::AbstractString; pagesize::Int=10, kwargs...)
+    lines = readlines(IOBuffer(text))
+    return Pager(lines, pagesize, 0, nothing, Config(; kwargs...))
+end
+
+function header(p::Pager)
+    total = length(p.lines)
+    current = min(p.pageoffset + p.pagesize, total)
+    percent = round(Int, (current / total) * 100)
+    return "($(lpad(current, ndigits(total))) / $total) $(lpad(percent, 3))%"
+end
+
+options(p::Pager) = p.lines
+
+cancel(::Pager) = nothing
+
+pick(::Pager, ::Int) = true
+
+function writeline(buf::IOBuffer, pager::Pager{Config}, idx::Int, iscursor::Bool)
+    print(buf, pager.lines[idx])
+end
+
+function writeLine(buf::IOBuffer, pager::Pager{<:Dict}, idx::Int, cursor::Bool)
+    cursor ? print(buf, pager.config[:cursor] ," ") : print(buf, "  ")
+    print(buf, pager.lines[idx])
+end
+
+function pager(terminal, object)
+    lines, columns = displaysize(terminal)
```

Comment:
```suggestion
    lines, columns = displaysize(terminal)::Tuple{Int,Int}
```
to protect against invalidation (`displaysize(terminal)` is not inferrable).